main/src/cgeo/geocaching/activity/OAuthAuthorizationActivity.java

```diff
@@ -63,47 +64,67 @@
     private ProgressDialog requestTokenDialog = null;
     private ProgressDialog changeTokensDialog = null;
 
-    private final Handler requestTokenHandler = new Handler() {
+    private final Handler requestTokenHandler = new RequestTokenHandler(this);
+    private final Handler changeTokensHandler = new ChangeTokensHandler(this);
+
+    private static final class RequestTokenHandler extends Handler {
+        private final WeakReference<OAuthAuthorizationActivity> activityRef;
```

Comment:
That's not possible, because `OAuthAuthorizationActivity` does not extend `Activity`, which is expected by `WeakReferenceHandler`.